pkg/sdk/testint/users_integration_test.go

```diff
@@ -161,68 +257,690 @@ func TestInt_UserCreate(t *testing.T) {
+					MultiStatementCount:                      sdk.Int(0),
+					NoorderSequenceAsDefault:                 sdk.Bool(false),
+					OdbcTreatDecimalAsInt:                    sdk.Bool(true),
+					QueryTag:                                 sdk.String("some_tag"),
+					QuotedIdentifiersIgnoreCase:              sdk.Bool(true),
+					RowsPerResultset:                         sdk.Int(2),
+					S3StageVpceDnsName:                       sdk.String("vpce-some_dns-vpce.amazonaws.com"),
+					SearchPath:                               sdk.String("$public, $current"),
+					SimulatedDataSharingConsumer:             sdk.String("some_consumer"),
+					StatementQueuedTimeoutInSeconds:          sdk.Int(10),
+					StatementTimeoutInSeconds:                sdk.Int(10),
+					StrictJSONOutput:                         sdk.Bool(true),
+					TimestampDayIsAlways24h:                  sdk.Bool(true),
+					TimestampInputFormat:                     sdk.String("YYYY-MM-DD"),
+					TimestampLTZOutputFormat:                 sdk.String("YYYY-MM-DD HH24:MI:SS"),
+					TimestampNTZOutputFormat:                 sdk.String("YYYY-MM-DD HH24:MI:SS"),
+					TimestampOutputFormat:                    sdk.String("YYYY-MM-DD HH24:MI:SS"),
+					TimestampTypeMapping:                     sdk.Pointer(sdk.TimestampTypeMappingLtz),
+					TimestampTZOutputFormat:                  sdk.String("YYYY-MM-DD HH24:MI:SS"),
+					Timezone:                                 sdk.String("Europe/Warsaw"),
+					TimeInputFormat:                          sdk.String("HH24:MI"),
+					TimeOutputFormat:                         sdk.String("HH24:MI"),
+					TraceLevel:                               sdk.Pointer(sdk.TraceLevelOnEvent),
+					TransactionAbortOnError:                  sdk.Bool(true),
+					TransactionDefaultIsolationLevel:         sdk.Pointer(sdk.TransactionDefaultIsolationLevelReadCommitted),
+					TwoDigitCenturyStart:                     sdk.Int(1980),
+					UnsupportedDDLAction:                     sdk.Pointer(sdk.UnsupportedDDLActionFail),
+					UseCachedResult:                          sdk.Bool(false),
+					WeekOfYearPolicy:                         sdk.Int(1),
+					WeekStart:                                sdk.Int(1),
+				},
+				ObjectParameters: &sdk.UserObjectParameters{
+					EnableUnredactedQuerySyntaxError: sdk.Bool(true),
+					NetworkPolicy:                    sdk.Pointer(networkPolicy.ID()),
+					PreventUnloadToInternalStages:    sdk.Bool(true),
+				},
+			},
+		}
+
+		err = client.Users.Alter(ctx, id, alterOpts)
+		require.NoError(t, err)
+
+		assertParametersSet(id)
+
+		// unset is split into two because:
+		// 1. this is how it's written in the docs https://docs.snowflake.com/en/sql-reference/sql/alter-user#syntax
+		// 2. current implementation of sdk.UserUnset makes distinction between user and session parameters,
+		// so adding a comma between them is not trivial in the current SQL builder implementation
+		alterOpts = &sdk.AlterUserOptions{
```

Comment:
I think such behavior should be validated and tested.

Reply:
Added a unit test to check unset validation (and set validation too)